Source/Core/DolphinQt2/CheatsManager.cpp

```diff
@@ -0,0 +1,342 @@
+    return sizeof(float);
+  case TYPE_DOUBLE_INDEX:
+    return sizeof(double);
+  default:
+    return m_match_value->text().toStdString().size();
+  }
+}
+
+template <typename T>
+bool Compare(u8* ptr, T value, int op)
+{
+  T mem_value = *reinterpret_cast<T*>(ptr);
+
+  switch (op)
+  {
+  case CMP_EQUAL_INDEX:
+    return mem_value == value;
+  case CMP_NOT_EQUAL_INDEX:
+    return mem_value != value;
+  case CMP_LESS_INDEX:
+    return mem_value < value;
+  case CMP_LESS_EQUAL_INDEX:
+    return mem_value <= mem_value;
+  case CMP_MORE_INDEX:
+    return value > mem_value;
+  case CMP_MORE_EQUAL_INDEX:
+    return value >= mem_value;
+  default:
+    return false;
+  }
+
+  return mem_value == value;
+}
+
+bool CheatsManager::MatchesSearch(u8* ptr) const
+{
+  const auto text = m_match_value->text();
+  const int op = m_match_operation->currentIndex();
+
+  const int base =
+      (m_match_decimal->isChecked() ? 10 : (m_match_hexadecimal->isChecked() ? 16 : 8));
+
+  switch (m_match_length->currentIndex())
+  {
+  case TYPE_BYTE_INDEX:
+    return Compare<u8>(ptr, text.toUShort(nullptr, base) & 0xFF, op);
+  case TYPE_SHORT_INDEX:
+    return Compare<short>(ptr, text.toUShort(nullptr, base), op);
+  case TYPE_INT_INDEX:
+    return Compare<u32>(ptr, text.toUInt(nullptr, base), op);
+  case TYPE_FLOAT_INDEX:
+    return Compare<float>(ptr, text.toFloat(), op);
+  case TYPE_DOUBLE_INDEX:
+    return Compare<double>(ptr, text.toFloat(), op);
+
+  case TYPE_STRING_INDEX:
+  {
+    std::string mem_text(reinterpret_cast<char*>(ptr), text.toStdString().size());
+
+    // String only supports equals and not equals comparisons because the other ones frankly don't
+    // make any sense here
+    switch (op)
+    {
+    case CMP_EQUAL_INDEX:
+      return mem_text == text.toStdString();
+    case CMP_NOT_EQUAL_INDEX:
+      return mem_text != text.toStdString();
+    default:
+      return false;
+    }
+  }
+  }
+
+  return false;
+}
+
+void CheatsManager::NewSearch()
+{
+  m_results.clear();
+  const u32 base_address = 0x80000000;
+
+  if (!Memory::m_pRAM)
+  {
+    m_result_label->setText(tr("Memory Not Ready"));
+    return;
+  }
+
+  const auto prev_state = Core::GetState();
+  Core::SetState(Core::State::Paused);
+
+  for (u32 i = 0; i < Memory::REALRAM_SIZE - GetTypeSize(); i++)
+  {
+    if (PowerPC::HostIsRAMAddress(base_address + i) && MatchesSearch(Memory::m_pRAM + i))
+      m_results.push_back(base_address + i);
+  }
+
+  Core::SetState(prev_state);
+
+  m_match_next->setEnabled(true);
+
+  Update();
+}
+
+void CheatsManager::NextSearch()
+{
+  const u32 base_address = 0x80000000;
+
+  if (!Memory::m_pRAM)
+  {
+    m_result_label->setText(tr("Memory Not Ready"));
+    return;
+  }
+
+  const auto prev_state = Core::GetState();
+  Core::SetState(Core::State::Paused);
+
+  for (auto it = m_results.begin(); it != m_results.end(); it++)
+  {
+    if (!PowerPC::HostIsRAMAddress(*it) || !MatchesSearch(Memory::m_pRAM + *it - base_address))
```

Comment:
This can utilize `remove_if`. This way you block delete all the non-matching entries in one pass instead of doing individual successive individual deletions.